src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -418,6 +395,39 @@ core::smart_refctd_ptr<IShaderCompiler::CCache> IShaderCompiler::CCache::deseria
     return retVal;
 }
 
+static void* SzAlloc(ISzAllocPtr p, size_t size) { p = p; return _NBL_ALIGNED_MALLOC(size, _NBL_SIMD_ALIGNMENT); }
+static void SzFree(ISzAllocPtr p, void* address) { p = p; _NBL_ALIGNED_FREE(address); }
+
+void nbl::asset::IShaderCompiler::CCache::SEntry::setContent(const asset::ICPUBuffer* uncompressedSpirvBuffer, dependency_container_t&& dependencies)
+{
+    dependencies = std::move(dependencies);
+    uncompressedContentHash = uncompressedSpirvBuffer->getContentHash();
+    uncompressedSize = uncompressedSpirvBuffer->getSize();
+
+    size_t propsSize = LZMA_PROPS_SIZE;
+    size_t destLen = uncompressedSpirvBuffer->getSize() + uncompressedSpirvBuffer->getSize() / 3 + 128;
+    std::vector<unsigned char> compressedSpirv = {};
+    compressedSpirv.resize(propsSize + destLen);
```

Comment:
why are you making a temp vector just to `memcpy` it later into a `spirv` buffer?

just make the buffer now and write into that

Reply:
the final size after compression is unknown and `ICPUBuffer` can't be resized (unless you ask me to add a `resize` method)?

Reply:
there's a thing called `CCPUBuffer` with a custom allocator that can also adopt memory

you can `malloc` instead of having a vector, then `realloc` and after realloc is complete you can create that `CCustomAllocatorCPUBuffer` with `core::adopt_memory` tag constructor overload

You just need to provide an "allocator" which will call `free()` on the adopted pointer

Reply:
i'll address this in the upcoming *Lib Shader Stage* PR

Reply:
erm since Lib Shader stage is blocked, copy or cherry pick the fix

Reply:
huh there's no `CCPUBuffer` in nabla (anymore?)